server/etcdserver/server.go

```diff
@@ -2414,8 +2423,10 @@ func (s *EtcdServer) snapshot(snapi uint64, confState raftpb.ConfState) {
 			"saved snapshot",
 			zap.Uint64("snapshot-index", snap.Metadata.Index),
 		)
-		s.storageVersionUpdated.Do(func() {
-			err := serverversion.UpdateStorageVersion(s.lg, s.be.BatchTx())
+		// Update storage schema after snapshot as fields introduced in v3.5 should be set.
+		// Remove in v3.7
```

Comment:
Nit and could we open an issue with 3.7 milestone label so we don't forget?

```suggestion
		// TODO: Remove in v3.7
```